File: cs50/pset1/credit.c
#include <cs50.h> // use get_long
#include <stdio.h>
#include <math.h>

int main(void)
{
    long n; //input
    long n_size = 0; //size of the input
    do
    {
        n = get_long("Number: ");
    }
    while (n < 0);

    long tempInput = n;
    while (tempInput != 0)
    {
        tempInput = tempInput / 10;
        n_size++;
    }


    if ((n_size) != 13 && (n_size) != 16 && (n_size) != 15)
    {
        printf("INVALID\n");
        return 0;
    }

    long arrayInput[n_size]; //array declare

    tempInput = n;
    for (long i = 0; i < n_size; i++)
    {
        arrayInput[n_size - i - 1] = tempInput % 10;
        tempInput = tempInput / 10;
    }


    long sum = 0;
    long counter = 0;
    long d1, d2;
    for (long j = 1; j < n_size; j += 2) // long sum1 *2
    {
        counter = arrayInput[n_size - j - 1] * 2; // long sum1 *2
        if (counter < 10)
        {
            d2 = counter;
            printf("%li\n", counter); // long sum1 < 10
        }
        else
        {
            d1 = counter % 10;
            counter = d1 + 1;
        }
        sum += counter;
    }
    printf("%li\n", sum);

    long summ = 0;
    long counterr = 0;
    for (long j = 0; j < n_size; j += 2)
    {
        counterr = arrayInput[n_size - j - 1];
        summ += counterr;  // long sum2

    }
    printf("%li\n", summ);


    long sumt = sum + summ;  // total sum = sum1+ sum2
    printf("%li\n", sumt);

    if ((n_size == 15 && arrayInput[0] == 3) && (arrayInput[1] == 4 || arrayInput[1] == 7) && (sumt % 10 == 0))
    {

        printf("AMEX\n"); // amex size:15 
    }

    else if ((n_size == 16 && arrayInput[0] == 5) && ((arrayInput[1] == 2 || arrayInput[1] == 1 || arrayInput[1] == 4
             || arrayInput[1] == 3 || arrayInput[1] == 5)) && (sumt % 10 == 0))
    {

        printf("MASTERCARD\n");

    }

    else if (((n_size == 13 && arrayInput[0] == 4) || (n_size == 16 && arrayInput[0] == 4)) && (sumt % 10 == 0))
    {

        printf("VISA\n");

    }
    else
    {
        printf("INVALID\n");
    }


}

File: cs50/pset6/credit.py
# TODO
import sys

from cs50 import get_int

# we get input from user
while True:
    try:
        n = get_int("Number: ")
        if (n > 0):
            break
    except:
        print("", end="")
# we change iint to string
y = str(n)

# credit card number lenght
if len(y) != 16 and len(y) != 13 and len(y) != 15:
    print("INVALID", end="")
    print()

    sys.exit()

# declare a numbers
t1, t2, sum = 0, 0, 0

for i in range(len(y) - 2, -1, -2):

    multi = int(y[i]) * 2
    if (multi < 10):
        t1 = multi

    # if multiplication is two digit and we need a remainder
    else:
        t2 = multi % 10
        multi = t2 + 1

    sum += multi


sum1 = 0
# no multi total
for i in range(len(y) - 1, -1, -2):
    sum1 += int(y[i])

total = sum1 + sum

if (total % 10 == 0):
    first_numb = int(y[0])
    second_numb = int(y[1])

    if first_numb == 3 and second_numb == 4 or second_numb == 7:
        print("AMEX")
    elif first_numb == 5 and 1 <= second_numb <= 5:
        print("MASTERCARD")
    elif first_numb == 4:
        print("VISA")
    else:
        print("INVALID")